core/src/main/java/com/graphhopper/routing/DefaultWeightingFactory.java

```diff
@@ -70,26 +71,46 @@ public Weighting createWeighting(Profile profile, PMap requestHints, boolean dis
             throw new IllegalArgumentException("You have to specify a weighting");
 
         Weighting weighting = null;
+        BooleanEncodedValue accessEnc = encodingManager.getBooleanEncodedValue(EncodingManager.getKey(vehicle, "access"));
+        DecimalEncodedValue speedEnc = encodingManager.getDecimalEncodedValue(EncodingManager.getKey(vehicle, "average_speed"));
+        DecimalEncodedValue priorityEnc = encodingManager.hasEncodedValue(EncodingManager.getKey(vehicle, "priority"))
+                ? encodingManager.getDecimalEncodedValue(EncodingManager.getKey(vehicle, "priority"))
+                : null;
         if (CustomWeighting.NAME.equalsIgnoreCase(weightingStr)) {
             if (!(profile instanceof CustomProfile))
                 throw new IllegalArgumentException("custom weighting requires a CustomProfile but was profile=" + profile.getName());
             CustomModel queryCustomModel = requestHints.getObject(CustomModel.KEY, null);
             CustomProfile customProfile = (CustomProfile) profile;
             queryCustomModel = CustomModel.merge(customProfile.getCustomModel(), queryCustomModel);
-            weighting = CustomModelParser.createWeighting(encoder, encodingManager, turnCostProvider, queryCustomModel);
+            // todonow: this used to be encoder.getMaxSpeed()!
+            double maxSpeed = speedEnc.getMaxDecimal();
```

Comment:
This is one place where we need the maximum actually used speed rather than the maximum storable speed.